products/extensions/search/How-to-define-your-custom-Elasticsearch-definition.md

```diff
@@ -0,0 +1,139 @@
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Elasticsearch\Framework\AbstractElasticsearchDefinition;
+use Shopware\Elasticsearch\Framework\ElasticsearchQueryHelper;
+
+class YourCustomElasticsearchDefinition extends AbstractElasticsearchDefinition
+{
+    public function __construct(
+        private readonly EntityDefinition $definition,
+        private readonly Connection $connection,
+        private readonly AbstractSearchLogic $searchLogic,
+        private readonly array $languageAnalyzerMapping
+    ) {
+    }
+
+    public function getMapping(Context $context): array
+    {
+        $languages = $this->connection->fetchAllKeyValue(
+            'SELECT LOWER(HEX(language.`id`)) as id, locale.code
+             FROM language
+             INNER JOIN locale ON locale_id = locale.id'
+        );
+
+        $languageFields = ElasticsearchQueryHelper::mapTranslatedField(
+            $languages,
+            $this->languageAnalyzerMapping,
+            self::getTextFieldConfig()
+        );
+
+        $properties = [
+            'id' => self::KEYWORD_FIELD,
+            'name' => $languageFields,
+            'description' => $languageFields,
+        ];
+
+        return [
+            '_source' => ['includes' => ['id']],
+            'properties' => $properties,
+        ];
+    }
+
+    public function buildTermQuery(Context $context, Criteria $criteria): BoolQuery
+    {
+        return $this->searchLogic->build($this->definition, $criteria, $context);
+    }
+
+    public function fetch(array $ids, Context $context): array
+    {
+        $data = $this->fetchData($ids, $context);
+
+        $documents = [];
+
+        foreach ($data as $id => $item) {
+            ['translation' => $translations] = ElasticsearchQueryHelper::parseJson($item, ['translation']);
+
+            $document = array_merge([
+                'id' => $id,
+            ], ElasticsearchQueryHelper::mapTranslatedFieldsValue([
+                'keywords',
+                'packUnit',
+                'packUnitPlural',
+            ], true, ...$translations));
+
+            $documents[$id] = $document;
+        }
+
+        return $documents;
+    }
+
+    public function getEntityDefinition(): EntityDefinition
+    {
+        return $this->definition;
+    }
+
+    private function fetchData(array $ids, Context $context): array
+    {
+        $sql = <<<'SQL'
+SELECT
+    LOWER(HEX(your_custom_table.id)) AS id,
+    #translation#
+FROM your_custom_table manufacturer
+    LEFT JOIN your_custom_table_translation ON your_custom_table_translation.your_custom_table_id = your_custom_table.id
+WHERE your_custom_table.id IN (:ids)
+GROUP BY your_custom_table.id
+SQL;
+
+        $sql = str_replace('#translation#', ElasticsearchQueryHelper::groupConcatSql('your_custom_table_translation', 'translation', ['name', 'description'], ['language_id']), $sql);
```

Comment:
as discussed in slack, that's to complicated. We should just "facade" the concat logic not the translation, aliases and UUID logic

```
SqlHelper::objectArray(
    [
        'id' => 'LOWER(HEX(tag.id))',
        'name' => 'tag.name'
    ],
    'tag_object'  // alias
);
```

Reply:
Hi @OliverSkroblin 
I am working on that, we will update it to the docs afterward